### PR TITLE
RUMM-1568 Add support to all regions

### PR DESCRIPTION
**Pull request by maxep**
### What and why?

Add support for regions: `us1`, `us3`, `us5`, `us1_fed`, `eu1`

### CODE REVIEW

File: Example/Podfile [hunk 1]
```diff
@@ -9,6 +9,7 @@ target 'DatadogSDKBridge_Tests' do
   # pod 'DatadogSDK', :path => '../../dd-sdk-ios/'
 
   pod 'DatadogSDKBridge', :path => '../'
+  pod 'DatadogSDK', :git => 'https://github.com/DataDog/dd-sdk-ios', :branch => 'master'
```
Comment (maxep) [inline Example/Podfile:12]: Podfile and Podfile.lock must be reverted before merging.

File: DatadogSDKBridge/Classes/Implementation/DdSdkImplementation.swift [hunk 2]
```diff
@@ -57,14 +57,18 @@ internal class DdSdkImplementation: DdSdk {
...
+        case "us1", "us":
+            _ = ddConfigBuilder.set(endpoint: .us1)
+        case "eu1", "eu":
+            _ = ddConfigBuilder.set(endpoint: .eu1)
+        case "us3":
+            _ = ddConfigBuilder.set(endpoint: .us3)
+        case "us5":
+            _ = ddConfigBuilder.set(endpoint: .us5)
+        case "us1_fed", "gov":
+            _ = ddConfigBuilder.set(endpoint: .us1_fed)
```
Comment (buranmert) [inline DatadogSDKBridge/Classes/Implementation/DdSdkImplementation.swift:69]: where do we document these? how can a new user know about those options?

**Review by buranmert — COMMENTED** [review]
lgtm but i'm concerned of discoverability

File: Example/Tests/DdSdkTests.swift [hunk 3]
```diff
@@ -114,35 +114,156 @@ internal class DdSdkTests: XCTestCase {
...
+            site: "EU1",
+            trackingConsent: "pending",
+            additionalConfig: nil
+        )
 
         let ddConfig = DdSdkImplementation().buildConfiguration(configuration: configuration)
 
-        XCTAssertEqual(ddConfig.datadogEndpoint, Datadog.Configuration.DatadogEndpoint.gov)
+        XCTAssertEqual(ddConfig.datadogEndpoint, .eu1)
     }
```
Comment (ncreated) [inline Example/Tests/DdSdkTests.swift:267]: nit: it's quite repetitive already, so to make it easier for next contributions we could introduce partial mock:
```swift
func testBuildConfigurationEU1Endpoint() {
   let configuration: DdSdkConfiguration = .mockWith(site: "EU1")
   let ddConfig = DdSdkImplementation().buildConfiguration(configuration: configuration)
   XCTAssertEqual(ddConfig.datadogEndpoint, .eu1)
}
```

**Review by ncreated — APPROVED** [review]

File: DatadogSDKBridge/Classes/Implementation/DdSdkImplementation.swift (hunk 2 above)
Comment (maxep) [reply DatadogSDKBridge/Classes/Implementation/DdSdkImplementation.swift:69]: Ah good catch, it should be documented [here](https://github.com/DataDog/dd-bridge-ios/blob/main/DatadogSDKBridge/Classes/Interface/DdSdkConfiguration.swift#L17)!

**Review by buranmert — APPROVED** [review]

**Review by ncreated — APPROVED** [review]

**Pull request merged**